lib/automaticupgrades/version.go

```diff
@@ -73,3 +72,60 @@ func Version(ctx context.Context, baseURL string) (string, error) {
 
 	return versionString, trace.Wrap(err)
 }
+
+// getVersionURL returns the versionURL or the default stable/cloud version url.
+func getVersionURL(versionURL string) (string, error) {
+	if versionURL != "" {
+		return versionURL, nil
+	}
+	cloudStableVersionURL, err := url.JoinPath(stableCloudVersionBaseURL, stableCloudVersionPath)
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+	return cloudStableVersionURL, nil
+}
+
+// Critical returns true if a critical upgrade is available.
+func Critical(ctx context.Context, criticalURL string) (bool, error) {
```

Comment:
This is the same as `Version()`, just fetches a different endpoint, let's factor out common logic into a separate method that both `Version()` and `Critical()` can use.